src/features/references/editor/ReferencesTableView.tsx

```diff
@@ -142,64 +134,75 @@ export function ReferencesTableView({ defaultFilter = '' }: { defaultFilter?: st
   const rowData = useMemo(() => references.map((r) => ({ ...r })), [references]);
 
   return (
-    <div className="flex w-full flex-col overflow-y-auto p-6">
-      {references.length === 0 && <UploadTipInstructions />}
-
-      <div className="relative mb-4 flex items-center justify-between gap-10">
-        <input
-          className="grow border border-slate-400 bg-slate-50 px-4 py-1 outline-none"
-          data-testid="filter-text-box"
-          id="filter-text-box"
-          placeholder="Search within references..."
-          ref={autoFocusAndSelect}
-          type="text"
-          value={quickFilter}
-          onInput={(e) => setQuickFilter(e.currentTarget.value)}
-        />
-        <div className="text-md flex items-center gap-2" data-testid="actions-menu">
-          <TopActionIcon
-            action="Add"
-            icon={VscNewFile}
-            onClick={() => emitEvent('refstudio://menu/references/upload')}
+    <div className="flex flex-1 flex-col items-stretch gap-6 pb-10">
+      <div className="flex items-center justify-center gap-4">
+        <div className="flex-1">
+          <SearchBar
+            autoFocus
+            fluid
+            initialValue={quickFilter}
+            placeholder="Filter author/title..."
+            onChange={(newValue) => setQuickFilter(newValue)}
           />
-          <VscKebabVertical />
-          <TopActionIcon
-            action="Remove"
+        </div>
+        <div className="h-8 w-px bg-content-area-border" />
+        <Button
+          Action={<AddIcon />}
+          size="M"
+          text="Add"
+          type="secondary"
+          onClick={() => emitEvent('refstudio://menu/references/upload')}
+        />
+        <div className="h-8 w-px bg-content-area-border" />
+        <div className="flex items-center justify-center gap-2">
+          <Button
+            Action={<BinIcon />}
             disabled={selectedReferences.length === 0}
-            icon={VscTrash}
-            selectedCount={selectedReferences.length}
+            size="M"
+            text={selectedReferences.length > 0 ? `Remove (${selectedReferences.length})` : 'Remove'}
+            type="secondary"
             onClick={handleOnBulkRemove}
           />
-          <TopActionIcon action="Export" disabled icon={VscDesktopDownload} />
-          <VscKebabVertical />
-          <TopActionIcon
-            action="Reset Cols"
-            icon={VscTable}
-            onClick={() => resetColumnsState(() => gridRef.current!.columnApi.resetColumnState())}
+          <Button
+            Action={<ExportIcon />}
+            size="M"
+            text="Export"
+            type="secondary"
+            onClick={() => emitEvent('refstudio://menu/references/export')}
           />
         </div>
+        <div className="h-8 w-px bg-content-area-border" />
+        <Button
```

Comment:
It seems odd that this is such a large button with such a small bit of functionality that only does anything if you've messed with the column widths. Not necessarily something to fix right now but worth thinking about?

Reply:
This is a good point @jvoytek I only added this initially because I was messing around a lot with columns. I agree that it can be smaller (ex: under a "3 vertical dots" menu popup).

Reply:
I'll share it with the design team